packages/runtime-core/src/rendererTemplateRef.ts

```diff
@@ -118,7 +118,11 @@ export function setRef(
           warn('Invalid template ref type:', ref, `(${typeof ref})`)
         }
       }
-      if (value) {
+
+      if (
+        value ||
+        (!value && vnode.transition && !vnode.transition.persisted)
```

Comment:
You shouldn't need the `!value` here. That should already be implicit if we're going into that branch of the conditional.

Reply:
OK